### DISCUSSION TITLE
Set multiple value using React Hook Form with Typescript

### DISCUSSION THREAD

**Discussion by 9hnd5** [react-hook-form/react-hook-form#6181] [Q&A]
Hi. Thank for you awesome library. I having a issue need your helping.

I have a form and I want to set value to this form at once. Using javascript it quite simple but when combine with typescript. I have no idea to workaround. Please see my code sandbox https://codesandbox.io/s/sweet-tu-wezfq?file=/src/App.tsx

Please give me some idea for this problem. I really appreciate it. Thanks

**Comment by jorisre** [comment]
Hi @huynd0902
I have a possible workaround by casting the value. `Object.entries` infer only `[string, T][]` or `[string, any][]` 

You can write your own `Entries` type:

```tsx
type Entries<T> = {
  [K in keyof T]: [K, T[K]];
}[keyof T][];
```

And use it like that:

```tsx
for (const [key, value] of Object.entries(formValue) as Entries<InputForm>) {
  setValue(key, value);
}
```

Source: https://stackoverflow.com/a/60142095/4973076

**Reply by 9hnd5** [reply]
Thanks @jorisre  for your solution. I'll try it. But while I waiting for any idea to workaround. I used the way below to solve my problem. I don't know it good or bad. Hope can someone point out.
```typescript
interface FormValue{
    firstName: string;
    lastName: string;
}
for (const [key, value] of Object.entries<any>(formValue)) {
    setValue(key as keyof FormValue, value);
}
```

**Comment by 9hnd5** [comment]
For those who still stuck on above problem. I have found a solution that I think it good for this case.
```typescript
interface FormValue{
    firstName: string;
    lastName: string;
}
const formValue: FormValue = {
    firstName: "xxx",
    lastName: "yyy"
}
for(const [key, value] of Object.entries(formValue) as Array<[keyof FormValue, FormValue<keyof FormValue>]>){
    setValue(key, value);
}
```
